### ISSUE TITLE
python 2 raise_from behaves incorrectly

### ISSUE THREAD

**Issue by mcarans** [benjaminp/six#193]
On Python 2, raise_from does not "raise from", it just raises. Hence, raise_from is not very useful and is quite confusing as it does not make Python 2 and 3 behave similarly.

I made something that worked better based on six's existing methods:
`
def raisefrom(exc_type, message, exc):
    if sys.version_info[:2] >= (3, 2):
        six.raise_from(exc_type(message), exc)
    else:
        six.reraise(exc_type, '%s - %s' % (message, exc), sys.exc_info()[2])
'

**Comment by benjaminp** [comment]
We can't really change the behavior of `raise_from` now, though maybe a new function would be okay. Only question is what to name it. :)

**Comment by mcarans** [comment]
That is indeed a tricky question. Did you come up with a name or were you waiting for me to suggest one? 

Whatever it is named will cause confusion because of the similarity of purpose, so I suggest raisefrom (no underscore) with good documentation to explain the difference possibly with a view to deprecating raise_from in the long term.

**Comment by benjaminp** [comment]
No, I don't have any great ideas off the top of my head. One aspect of your suggested implementation is that it only works within an `except` block. That needs to be clear to users.

**Comment by mcarans** [comment]
I think better not to call existing functions which may change but implement fresh like this:

```
if sys.version_info[:2] == (3, 2):
    exec_("""def raisefrom(exc_type, message, from_value):
    value = exc_type(message)
    try:
        if from_value is None:
            raise value
        raise value from from_value
    finally:
        exc_type = None
""")
elif sys.version_info[:2] > (3, 2):
    exec_("""def raisefrom(exc_type, message, from_value):
    value = exc_type(message)
    try:
        raise value from from_value
    finally:
        exc_type = None
""")
else:
    exec_("""def raisefrom(exc_type, message, from_value):
    try:
        raise exc_type, '%s - %s' % (message, from_value), sys.exc_info()[2]
    finally:
        exc_type = None
""")
```

**Comment by mcarans** [comment]
@benjaminp I have made a PR for the new function: https://github.com/benjaminp/six/pull/217
